3.4 clean up KernelIT

fickludd opened the pull request:
I found this old test-class called KernelIT, which contained lot's on test duplicates and other strange things. Cleaned it up a bit.

fickludd committed:
Improve test name.

fickludd committed:
Remove duplicated API mix test.

fickludd committed:
Moved rollback-tests on Kernel API level into Kernel API module

fickludd committed:
Move some Kernel API tests from neo4j-lucene-index to neo4j-kernel

fickludd committed:
Remove duplicates of NodeTransactionStateTestBase.shouldSeeLabelChangesInTransaction

fickludd committed:
Migrate assertion to NodeTransactionStateTestBase.shouldSeeLabelChangesInTransaction

fickludd committed:
Migrate assertion to NodeWriteTestBase

fickludd committed:
Remove duplicates of NodeWriteTestBase tests.

fickludd committed:
Migrate nodeRemoveLabel result assertion

fickludd committed:
Delete test that is no longer needed due to API changes

fickludd committed:
Migrate node delete test in nodeLabelScan to NodeLabelIndexCursorTestBase

fickludd committed:
Remove imports

SaschaPeukert commented on community/kernel/src/test/java/org/neo4j/kernel/impl/api/integrationtest/KernelIT.java:
```diff
@@ -78,394 +66,11 @@ public void mixingBeansApiWithKernelAPI() throws Exception
-        ktx.close();
-
         // 5: Commit through the beans API
         transaction.success();
```
Now the enumeration in the comments looks funny, because 4 is missing

SaschaPeukert commented on community/kernel-api/src/test/java/org/neo4j/internal/kernel/api/NodeLabelIndexCursorTestBase.java:
```diff
@@ -36,17 +37,25 @@
+            toDelete.delete();
+            tx.success();
+        }
     }
```
What is the reason for that change? To have an id that is known for being empty later?

fickludd replied:
To make sure that the label-scan doesn't return deleted nodes, it was in one of the `KernelIT` tests.

SaschaPeukert committed:
Fixed small test

Test will now add and remove a new label instead of removing the wrong label that was meant to stay there

SaschaPeukert reviewed (commented):
Nice! I just have a question and a nitpick
